src/message_validation.md

```diff
@@ -9,38 +9,49 @@
-     - Signature verification: Requires the AS (potentially blocking & persistence step might be required)
-     - subsequent steps depend on content_type
- - Content type: Application
- - Content type: Proposal
-   - Semantic checks:
-     - Add Proposal: Double join
-     - Remove Proposal: Ghost removal
-     - Update Proposal: Identity must be unchanged
-   - Operations like Add, Remove must be validated by AS: (potentially blocking & persistence step might be required)
- - Content type: Commit
-   - all proposals (either inline or referenced) must be valid (see Content type: Proposal) potentially blocking & persistence step might be required)
-   - Commit must not cover inline self Remove proposal
-   - Path must be present, unless Commit only covers Add Proposals
-   - Path must be the right length
-   - Staging step: proposals must be applied to modify the provisional tree
-   - Path must be applied and decrypt correctly
-   - New public keys from Path must be verified and match the private keys from the direct path
-   - Confirmation tag must be successfully verified
+### Semantic validation of message framing
+
+| ValidationStep | Description                                                 | Implemented | Tested | Test File                                    |
+| -------------- | ----------------------------------------------------------- | ----------- | ------ | -------------------------------------------- |
+| `ValSem1`      | Wire format                                                 | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem2`      | Group id                                                    | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem3`      | Epoch                                                       | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem4`      | Sender: Member: check the sender points to a non-blank leaf | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem5`      | Application messages must use ciphertext                    | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem6`      | Ciphertext: decryption needs to work                        | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem7`      | Membership tag presence                                     | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem8`      | Membership tag verification                                 | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem9`      | Confirmation tag presence                                   | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
+| `ValSem10`     | Signature verification                                      | ✅          | ✅     | `openmls/src/group/tests/test_validation.rs` |
```

Comment:
Turns out that github markdown doesn't support putting checkboxes in tables so I used emoji, feel free to suggest an alternative.

Reply:
These emojis look good to me. If it turns out that they can't be rendered at some point, we can always change them.